File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ListPayments.kt
package com.example.kotlinmpplib

import breez_sdk_liquid.*
class ListPayments {
    fun listPayments(sdk: BindingLiquidSdk) {
        // ANCHOR: list-payments
        try {
            val payments = sdk.listPayments()
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments
    }

}

File: snippets/rust/src/list_payments.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_liquid::prelude::*;

async fn list_payments(sdk: Arc<LiquidSdk>) -> Result<Vec<Payment>> {
    // ANCHOR: list-payments
    let payments = sdk.list_payments().await?;
    // ANCHOR_END: list-payments

    Ok(payments)
}
